File: src/main/java/info/idgst/digest/DigestRepository.java
package info.idgst.digest;

import org.springframework.data.repository.PagingAndSortingRepository;

/**
 * Repository for retrieving {@link Digest}.
 *
 * @author Aliaksei Bahdanau
 */
public interface DigestRepository extends PagingAndSortingRepository<Digest, String> {

    /**
     * Find {@link Digest} by specified title.
     *
     * @param title {@link Digest}'s title to find by.
     * @return {@link Digest}
     */
    Digest findByTitle(String title);
}

File: src/main/kotlin/info/devbug/digest/repository/DigestRepository.kt
package info.devbug.digest.repository

import org.springframework.data.repository.PagingAndSortingRepository
import org.springframework.stereotype.Repository

/**
 * @author Aliaksei Bahdanau
 */

@Repository
interface DigestRepository : PagingAndSortingRepository<DigestDto, Int> {

    fun findByTitle(title: String): DigestDto?

    /**
     * Finds [DigestDto] by specified ID.
     *
     * @return [DigestDto]
     */
    fun findById(id: String): DigestDto?
}